### PR TITLE
EventFacade tests (potential PII leaks)

### PR DESCRIPTION
**Pull request by Morpheu5**
Testing the following endpoints:

- `GET /events`
- `GET /events/bookings/{booking_id}`
- `POST /events/{event_id}/booking`
- `DELETE /events/{event_id}/bookings/cancel`
- `GET /events/{event_id}/bookings`
- `GET /events/{event_id}/groups_booking`
- `GET /events/{event_id}/bookings/for_group/{group_id}`

---

**Pull Request Check List**
- [ ] Unit Tests & Regression Tests Added (Optional)
- [x] E2E Tests Added
- [x] Removed Unnecessary Logs/System.Outs/Comments/TODOs
- [x] Security - Data Exposure - PII is not stored or sent unencrypted
- [x] Security - Data Exposure - Test any altered or created endpoints using [swagger](http://localhost:8080/isaac-api/api-docs/)
- [x] Security - Access Control - Check authorisation on every new endpoint
- [ ] Peer-Reviewed

### CODE REVIEW

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java [hunk 1]
```diff
@@ -44,66 +49,237 @@ public void setUp() {
...
+        assertEquals(UserSummaryWithEmailAddressDTO.class.getCanonicalName(), ((EventBookingDTO) getEventBookingsByIdResponse.getEntity()).getUserBooked().getClass().getCanonicalName());
+
+        // --- Delete the booking created above otherwise the other tests may be affected.
+        HttpServletRequest cancelBookingRequest = createRequestWithCookies(new Cookie[] { studentLogin.cookie });
+        replay(cancelBookingRequest);
+        Response cancelBookingResponse = eventsFacade.cancelBooking(cancelBookingRequest, "_regular_test_event");
+        assertEquals(Response.Status.NO_CONTENT.getStatusCode(), cancelBookingResponse.getStatus());
+
+        // --- Tear down
+        // This should not be necessary, but we don't actually remove cancelled bookings from the database.
```
Comment (mwtrew) [inline src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java:123]: There are a couple of comments about things which should be unnecessary but are here nonetheless, which is a bit confusing - it would be good to remove either the unnecessary thing or the comment.

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java [hunk 2]
```diff
@@ -44,66 +49,237 @@ public void setUp() {
...
+        // Alice is not associated with Dave but is booked for this event => Alice should not be present
+        assertFalse(daveAlice.isPresent());
+        Optional<UserSummaryDTO> daveCharlie = (Optional<UserSummaryDTO>) daveEntity.stream().filter(e -> ((EventBookingDTO) e).getUserBooked().getId() == 9).findFirst();
+        // Charlie is associated with Dave and is not booked for this event => Charlie should not be present
+        assertFalse(daveCharlie.isPresent());
+    }
+
+    @Test
+    // GET /events/{event_id}/bookings/for_group/{group_id} -> EventFacade::getEventBookingForGivenGroup(request, eventId, groupId)
+    public void getEventBookingsByEventIdForGroup() throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException, AuthenticationProviderMappingException, IncorrectCredentialsProvidedException, AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException, MFARequiredButNotConfiguredException {
```
Comment (mwtrew) [inline src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java:243]: [In the last PR](https://github.com/isaacphysics/isaac-api/pull/452#discussion_r930003423) you mentioned you tried a few ways of structuring the tests and settled on this. My preference would be more similar to how our unit tests are (should be?) structured - that is, one test method per behaviour in a particular context (i.e. using endpoint A user B can do C when D) with clear arrange/act/assert stages.

That might result in some duplication and may be inconvenient for tests which build up cumulative state, but I think it would be more obvious what's happening.

**Review by mwtrew — COMMENTED** [review]
These tests look good, if occasionally a bit hard to follow - I've made a couple of suggestions.

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java (hunk 1 above)
Comment (Morpheu5) [reply src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java:123]: The unnecessary thing is actually necessary. It would be unnecessary if we actually removed the record from the database, but we don't, and this leaves state hanging around confusing other test cases. I might remove the comment, but leave a remark so we are aware of why the tear down phase is there.

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java (hunk 2 above)
Comment (Morpheu5) [reply src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java:243]: Yes, I've been wrestling with my decision ever since and I haven't settled fully on this style. I'm tempted to leave these tests as they are for the moment and we can try the other approach going forward, see how it feels. The risk is that we mix styles but I'll try to keep on top of this and monitor the situation.
